src/App.tsx

```diff
@@ -1,203 +1,139 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { ErrorMessage, FilterStatus, Todo } from './types/Todo';
+import cn from 'classnames';
+import { getTodos } from './api/todos';
+import { TodoList } from './api/components/TodoList';
+import { HeaderTodoApp } from './api/components/HeaderTodoApp';
+import { FooterTodoApp } from './api/components/FooterTodoApp';
+
+const filterTodo = (todos: Todo[], filterStatus: FilterStatus) => {
+  switch (filterStatus) {
+    case FilterStatus.COMPLETED:
+      return todos.filter(todo => todo.completed);
+    case FilterStatus.ACTIVE:
+      return todos.filter(todo => !todo.completed);
+    default:
+      return todos;
+  }
+};
+
+export const findTodoById = (todos: Todo[], id: number) => {
+  return todos.find(todo => todo.id === id);
+};
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>(
+    ErrorMessage.DEFAULT,
+  );
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.DEFAULT);
+
+  const isAllChecked = useRef(false);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.TODO_LOAD);
+        setTimeout(() => {
+          setErrorMessage(ErrorMessage.DEFAULT);
+        }, 3000);
+      });
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return filterTodo(todos, filterStatus);
+  }, [todos, filterStatus]);
+
+  const handleAddTodo = (newTodo: Todo) => {
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleUpdateTodo = (updatedTodo: Todo) => {
+    setTodos(
+      todos.map(todo => (todo.id === updatedTodo.id ? updatedTodo : todo)),
+    );
+  };
+
+  const handleOnChangeTodoStatus = (id: number) => {
+    const currentTodo = findTodoById(todos, id);
+
+    if (currentTodo) {
+      currentTodo.completed = !currentTodo.completed;
+      setTodos(
+        todos.map(todo => (todo.id === currentTodo.id ? currentTodo : todo)),
+      );
+    }
+  };
+
+  const handleCheckAll = () => {
+    if (todos.every(todo => todo.completed)) {
+      isAllChecked.current = false;
+    } else {
+      isAllChecked.current = true;
+    }
+
+    setTodos(todos.map(todo => ({ ...todo, completed: isAllChecked.current })));
```

Comment:
```suggestion
const x = todos.map(todo => ({ ...todo, completed: isAllChecked.current }));
    setTodos(x);
```
x - should be named somehow, I don't understand what happend here (P.S i understand but... )  ;)